src/components/common/UserProfileEdit/UserAvatarUploader.tsx

```diff
@@ -28,49 +27,58 @@ const UserAvatarUploader = ({
   user,
   user: { walletAddress, profile },
 }: Props) => {
-  const userProfile = excludeTypenameKey(profile);
   const { updateUser } = useAppContext();
+  const [avatar, setAvatar] = useState<string | null>(profile?.avatar || null);
   const [updateAvatar, { error }] = useUpdateUserProfileMutation();
 
   const handleUpload = async (fileData: FileReaderFile) => {
     const file = await updateAvatar({
       variables: {
         input: {
           id: walletAddress,
-          ...userProfile,
           avatar: fileData.data,
         },
       },
     });
-    updateUser?.(user.walletAddress);
+    setAvatar(fileData.data);
     return file;
   };
 
   const remove = async () => {
     await updateAvatar({
       variables: {
-        input: { id: walletAddress, ...userProfile, avatar: null },
+        input: { id: walletAddress, avatar: null },
       },
     });
-    updateUser?.(user.walletAddress);
+    setAvatar(null);
   };
 
+  // on dismount
+  useEffect(
+    () => () => updateUser?.(user.walletAddress),
```

Comment:
Moving the call to `updateUser` to the component dismount avoids the `loadingSpinner` from displaying after uploading a new avatar. In the meantime we set the avatar to component state so it can be displayed as the placeholder for the `AvatarUploader`.